src/rpc/masternode.cpp

```diff
@@ -566,48 +566,43 @@ static RPCHelpMan masternodelist_helper(bool is_composite)
+                                    PadString(ToString(dmnToLastPaidTime(dmn)), 10),
+                                    PadString(ToString(dmn.pdmnState->nLastPaidHeight), 6),
+                                    dmn.pdmnState->addr.ToStringAddrPort());
+            if (!strFilter.empty() && strFull.find(strFilter) == std::string::npos &&
                 strOutpoint.find(strFilter) == std::string::npos) return;
             obj.pushKV(strOutpoint, strFull);
         } else if (strMode == "info") {
-            std::ostringstream streamInfo;
-            streamInfo << std::setw(18) <<
-                           dmnToStatus(dmn) << " " <<
-                           dmn.pdmnState->nPoSePenalty << " " <<
-                           payeeStr << " " <<
-                           dmn.pdmnState->addr.ToStringAddrPort();
-            std::string strInfo = streamInfo.str();
-            if (strFilter !="" && strInfo.find(strFilter) == std::string::npos &&
+            std::string strInfo = strprintf("%s %s %s %s",
+                                    PadString(dmnToStatus(dmn), 18),
+                                    dmn.pdmnState->nPoSePenalty,
+                                    payeeStr,
+                                    dmn.pdmnState->addr.ToStringAddrPort());
+            if (!strFilter.empty() && strInfo.find(strFilter) == std::string::npos &&
                 strOutpoint.find(strFilter) == std::string::npos) return;
             obj.pushKV(strOutpoint, strInfo);
         } else if (strMode == "json" || strMode == "recent" || strMode == "evo") {
-            std::ostringstream streamInfo;
-            streamInfo <<  dmn.proTxHash.ToString() << " " <<
-                           dmn.pdmnState->addr.ToStringAddrPort() << " " <<
-                           payeeStr << " " <<
-                           dmnToStatus(dmn) << " " <<
-                           dmn.pdmnState->nPoSePenalty << " " <<
-                           dmnToLastPaidTime(dmn) << " " <<
-                           dmn.pdmnState->nLastPaidHeight << " " <<
-                           EncodeDestination(PKHash(dmn.pdmnState->keyIDOwner)) << " " <<
-                           EncodeDestination(PKHash(dmn.pdmnState->keyIDVoting)) << " " <<
-                           collateralAddressStr << " " <<
-                           dmn.pdmnState->pubKeyOperator.ToString();
-            std::string strInfo = streamInfo.str();
-            if (strFilter !="" && strInfo.find(strFilter) == std::string::npos &&
+            std::string strInfo = strprintf("%s %s %s %s %s %s %s %s %s %s %s",
+                                    dmn.proTxHash.ToString(),
+                                    dmn.pdmnState->addr.ToStringAddrPort(),
+                                    payeeStr,
+                                    dmnToStatus(dmn),
+                                    dmn.pdmnState->nPoSePenalty,
+                                    ToString(dmnToLastPaidTime(dmn)),
```

Comment:
no need for `ToString()` here, could simply use `%d` in format

Reply:
Resolved in latest push!